src/broadcast/spectateManager.ts

```diff
@@ -131,23 +98,25 @@ export class SpectateManager extends EventEmitter {
     await new Promise<void>((resolve, reject) => {
       const socket = new WebSocketClient();
 
-      socket.on("connectFailed", (error) => {
-        log.error(`[Spectate] WS connection failed\n`, error);
-        this.emit(SpectateManagerEvent.ERROR, error);
+      socket.on("connectFailed", (err) => {
+        const errMsg = err.message || JSON.stringify(err);
+        this.emit(SpectateEvent.LOG, `WS connection failed\n${errMsg}`);
+        this.emit(SpectateEvent.ERROR, errMsg);
```

Comment:
Can we use the `log` event for logs and the `error` event for errors? It seems like in a lot of places we're doing the double emit to log and error the same message.